### ISSUE TITLE
byFunction skips anchors

### ISSUE THREAD

**Issue by bebraw** [sorensen/absolutify#3]
By the looks of it, `byFunction` doesn't give access to urls containing anchors.

**Demo:**

``` javascript
var _url = require('url');
var absolutify = require('absolutify');

var content = '<a href="#demo">demo</a></p>';

console.log(resolveUrls(content));

function resolveUrls (content) {
  return absolutify(content, function (url, attrName) {
    console.log('url', url);

    return url;
  });
}
```

It doesn't call that callback at all. I would expect it to given it's a valid `a`. I would need to shape the url to something else (RSS usage).

**Comment by sorensen** [comment]
@bebraw I had not considered manipulating anchors at all for this module, I can try and add support for this behavior, but it may take some time to figure out the RegExp.

**Comment by bebraw** [comment]
The RegExp is tricky, yeah. Maybe anchors could be considered as a case of their own somehow.

**Comment by sorensen** [comment]
I _think_ I may be onto something, was able to add `(?=#)` to the RegExp to find the anchors, testing now to see if that breaks anything else.

**Comment by bebraw** [comment]
Awesome. Thanks. 👍

**Comment by sorensen** [comment]
@bebraw I've published this at version `0.1.0`

**Comment by bebraw** [comment]
Super. 💯
